common/types/head_tracker.go

```diff
@@ -57,12 +60,17 @@ type HeadListener[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
 // NewHeadHandler is a callback that handles incoming heads
 type NewHeadHandler[H Head[BLOCK_HASH], BLOCK_HASH Hashable] func(ctx context.Context, header H) error
 
+// HeadBroadcaster relays heads from the head tracker to subscribed jobs, it is less robust against
```

Comment:
>>less robust than head tracker, and missed heads should be expected
That seems confusing and misleading.

Just say this:

HeadBroadcaster relays new Heads to all subscribers.